### PR TITLE
Simplify detection of the MAVEN_HOME and make it fully qualified

### PR DESCRIPTION
**Pull request by rosti-il**
The removed 'stripMHome' recursive function is not needed anymore.

Part of the 'mvn -version' output before this change:
Maven home: D:\develop\apache-maven-3.6.3\bin\..

Same part of the 'mvn -version' output after this change:
Maven home: D:\develop\apache-maven-3.6.3
I.e. now it's fully qualified.

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by slachiewicz** [comment]
Thank you for contribution - could you also create Jira here: https://issues.apache.org/jira/browse/MNG ?

**Comment by rosti-il** [comment]
Created a Jira issue about this improvement:
https://issues.apache.org/jira/browse/MNG-6837

**Comment by slachiewicz** [comment]
https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=488a555a9796b64e926daf2fde076e6cf4492683

**Comment by rosti-il** [comment]
I want to change this line in my commit:
`set "MAVEN_HOME=%MAVEN_HOME:\bin\=%"`

into this one:
`set "MAVEN_HOME=%MAVEN_HOME:~0,-5%"`

The reason: it's safer because theoretically the "/bin/" substring could appear more than once but we need to remove only the last one.

I will amend my commit and push force it.

**Comment by slachiewicz** [comment]
Quick question - what form where we use `"` should be better?
`set "MAVEN_HOME=xxxx"` or `set MAVEN_HOME="xxxx"`

**Comment by rosti-il** [comment]
I just kept the original style of using `"` in `set` commands. If you think another style is better I've no objection but this is another change that should be done on the whole `mvn.cmd` script, not only in set of the MAVEN_HOME. I think it should be different Jira and pull request anyway.

**Comment by rosti-il** [comment]
I just kept the original style of using `"` in `set` commands. But now I found that this is not only a matter of style but a mandatory requirement. Using `set A="value with spaces"` inserts the double quotes into the variable value and we don't want this. Using `set "A=value with spaces"` doesn't insert the double quotes into the variable value and this is good.

https://stackoverflow.com/questions/535975/dealing-with-quotes-in-windows-batch-scripts

File: apache-maven/src/bin/mvn.cmd [hunk 1]
```diff
@@ -60,15 +60,11 @@ echo NB: JAVA_HOME should point to a JDK not a JRE >&2
 goto error
 
 :chkMHome
-set "MAVEN_HOME=%~dp0.."
-if not "%MAVEN_HOME%"=="" goto stripMHome
+set "MAVEN_HOME=%~dp0"
+set "MAVEN_HOME=%MAVEN_HOME:~0,-5%"
```
Comment (michael-o) [inline apache-maven/src/bin/mvn.cmd]: This cannot be combined into one `set`?

**Pull request closed without merging**